theseus/embodied/robot/robot.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from typing import List, Dict, Optional
+import urdf_parser_py.urdf as urdf
+import torch
+
+from theseus.geometry.functional import so3, se3
+from .joint import Joint, FixedJoint, RevoluteJoint, PrismaticJoint
+from .link import Link
+
+
+class Robot(abc.ABC):
+    def __init__(self, name: str, dtype: torch.dtype = None):
+        if dtype is None:
+            dtype = torch.get_default_dtype()
+        self._name: str = name
+        self._dtype: torch.dtype = dtype
+        self._dof: int = 0
+        self._num_joints: int = 0
+        self._num_links: int = 0
+        self._joints: List[Joint] = []
+        self._links: List[Link] = []
+        self._joint_map: Dict[str, Joint] = {}
+        self._link_map: Dict[str, Link] = {}
+
+    @classmethod
+    def from_urdf_file(cls, urdf_file: str, dtype: torch.dtype = None) -> "Robot":
```

Comment:
Trailing `.._file` could be dropped from the name.

Reply:
Maybe `def from_urdf(fname: str)`?

Reply:
I remember in ROS, there are methods called `from_urdf()` and `from_urdf_file()` while the input from the former is a `urdf_model`.
<img width="444" alt="Screenshot 2023-02-28 at 7 03 31 PM" src="https://user-images.githubusercontent.com/6612911/222010933-d68fa900-24c8-49fd-bb5c-0a280df2a0f1.png">